meshroom/core/graph.py

```diff
@@ -768,7 +768,11 @@ def addNewNode(self, nodeType, name=None, position=None, **kwargs):
         if name and name in self._nodes.keys():
             name = self._createUniqueNodeName(name)
 
-        n = self.addNode(Node(nodeType, position=position, **kwargs), uniqueName=name)
+        # Get the Node Constructor which should be initialized for the given node type
+        # Node or Backdrop...
+        NodeType = getPreferredNodeConstructor(nodeType)
+
+        n = self.addNode(NodeType(nodeType, position=position, **kwargs), uniqueName=name)
```

Comment:
Directly use a function to create the node, instead of a function to return the node type.
```suggestion
        n = self.addNode(createNode(nodeType, position=position, **kwargs), uniqueName=name)
```

Reply:
This has been updated.